cpp/src/arrow/flight/client.cc

```diff
@@ -34,6 +34,9 @@
 #include <grpc++/grpc++.h>
 #endif
 
+#include "grpc/grpc_security_constants.h"
+#include "grpcpp/security/tls_credentials_options.h"
```

Comment:
External includes, thus no `"`
```suggestion
#include <grpc/grpc_security_constants.h>
#include <grpcpp/security/tls_credentials_options.h>
```